### PR TITLE
chore(ci): add xunit output to evergreen config

### PR DESCRIPTION
**Pull request by emadum**
## Description

- Adds xunit reporting to the `check:` package scripts, via the `spec-xunit-file` reporter. This reporter is useful because it continues to send the output of the default `spec` reporter to stdout, while also generating an XML file with the xunit results. 
- Updated evergreen config to upload these results via `attach.xunit_results` in a `post` hook
- This shows individual test results in Evergreen, [example here](https://spruce.mongodb.com/task/mongo_node_driver_ubuntu_14.04_dubnium_test_2.6_replica_set_patch_ba76696ef56629b7aba60cbb27ef1d14f7e06897_5f9751aa30661541eb0b4316_20_10_26_22_46_30/tests?execution=1) - makes it a lot easier to see specific failing tests at a glance.

**What changed?**

**Are there any files to ignore?**

### CODE REVIEW

File: test/manual/mocharc.json [hunk 1]
```diff
@@ -1,3 +1,4 @@
 {
-  "require": "ts-node/register"
+  "require": "ts-node/register",
+  "reporter": "spec-xunit-file"
```
Comment (mbroadst) [inline test/manual/mocharc.json:3]: hm this is specified in the script above, can we reduce it to one place?

**Review by mbroadst — APPROVED** [review]
LGTM

File: .evergreen/config.yml.in [hunk 2]
```diff
@@ -103,7 +103,7 @@ functions:
       params:
         file: mo-expansion.yml
 
-  bootstrap mongohoused:
+  "bootstrap mongohoused":
```
[the remark this answers is not in the document]
Comment (emadum) [reply .evergreen/config.yml.in:106]: Yeah, I just added quotes to a few that were missing them for consistency - it makes the titles show up in a different color on syntax highlighting, which I thought was nice, but I can revert that.

File: test/manual/mocharc.json (hunk 1 above)
Comment (emadum) [reply test/manual/mocharc.json:3]: I didn't add `--report spec-xunit-file` to the npm scripts that use this file via the `--config` option in order to reduce verbosity. I can expand it out into the package scripts if desired.

File: .evergreen/config.yml [hunk 3]
```diff
@@ -207,7 +207,6 @@ functions:
       params:
         working_dir: src
         script: |
-          ${PREPARE_SHELL}
```
[the remark this answers is not in the document]
Comment (emadum) [reply .evergreen/config.yml:210]: I'm guessing it was a change that was made directly to the `config.yml` and not to `config.yml.in`, so it was lost when regenerating the file. I can take a closer look.

File: test/manual/mocharc.json (hunk 1 above)
Comment (nbbeeken) [reply test/manual/mocharc.json:3]: Can't you add this same setting to `.mocharc.json`?

**Review by nbbeeken — APPROVED** [review]
My logs filter PR #2605 undoes the work here in favor of an in house xml reporter.
There's a few features that this xml reporter lacks:
- it doesn't capture stdout and stderr (meaning you can see deprecation warnings, for tests that care about that, or run special patches with print statements for debugging)
- it escapes the text inside the CDATA blocks which is unnecessary, it makes reading the errors a bit strange
- the timestamps aren't an ISO format
- it emits one testsuite for all mocha tests, it doesn't change much about the display but its logically inconsistent with our test organization

I'm still LGTM-ing this though because all this isn't to say we shouldn't consider merging this, my PR is larger, and comes with more mental overhead, if we want this test reporting feature right now (which it is really nice to have) and end up accepting my PR later I still think that that could be worth our while.

File: .evergreen/config.yml.in (hunk 2 above)
[the remark this answers is not in the document]
Comment (emadum) [reply .evergreen/config.yml.in:106]: @mbroadst Just confirming, should I revert the quotation mark changes I made to improve consistency?

[the remark this answers is not in the document]
Comment (mbroadst) [reply .evergreen/config.yml.in:106]: No that's fine, I just wondered what was going on

**Pull request merged**